### ISSUE TITLE
Yh

### ISSUE THREAD

**Issue by SRTAI22** [SRTAI22/NeuralNet_KMINST#1]
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
import torchvision
import torchvision.transforms as transforms
import matplotlib.pyplot as plt
import requests
import spicy

device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')

learning_rate = 0.001
batch_size = 256
input_size = 60000
hidden_size = 100         
num_classes = 101

my_transform = transforms.Compose([transforms.Resize((200,300)), transforms.ToTensor()])

train_dataset = torchvision.datasets.Caltech101(root='./dataCT', download=True, transform=my_transform)

test_dataset = torchvision.datasets.Caltech101(root='./dataCT', transform=my_transform)

train_loader = torch.utils.data.DataLoader(dataset=train_dataset, batch_size=batch_size, shuffle=True)

test_loader = torch.utils.data.DataLoader(dataset=test_dataset, batch_size=batch_size, shuffle=False)

class NeuralNet(nn.Module):
    def __init__(self, input_size, hidden_size, num_classes):
        super(NeuralNet, self).__init__()
        self.input_size = input_size
        self.l1 = nn.Linear(input_size, hidden_size)
        self.relu = nn.LeakyReLU()()
        self.l2 = nn.Linear(hidden_size, num_classes)

    def forward(self,x):
        out = self.l1(x)
        out = self.relu(out)
        out = self.l2(out)

        return out

model =  NeuralNet(input_size, hidden_size, num_classes).to(device)

l = nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate)

# loop over the dataset multiple times

n_t_s = len(train_loader)

for epoch in range(5):
    for i, (images, labels) in enumerate(train_loader, 0):
        images = images.reshape(-1, 200*300).to(device)
        labels = labels.to(device)

        ot = model(images)
        loss = l(ot, labels)

        optimizer.zero_grad()
        loss.backward()
        optimizer.step()

        if(i+1) % 100 == 0:
            print(f' epoch : {epoch+1}/{num_epoch}, step : {i+1}/{n_t_s}, loss : {loss.item():.4f}')

print('Finished Training')
